src/extension/app-schema/app-schema-test/pom.xml

```diff
@@ -163,6 +163,14 @@
             <version>2.0.0-1</version>
             <scope>test</scope>
         </dependency>
+        <!-- required for Web UI tests -->
+        <dependency>
+            <groupId>org.geoserver.web</groupId>
+            <artifactId>gs-web-demo</artifactId>
+            <version>${project.version}</version>
+            <classifier>tests</classifier>
+            <scope>test</scope>
+        </dependency>
```

Comment:
@bencaradocdavies Ouch, looks like I forgot to check the Maven build.

I pushed another commit to fix the build, but refrained from squashing it and rebasing the branch (I remember you advised against it): I can do it in a new branch and submit a new PR, if you like.